admin/controllers/school.js

```diff
@@ -449,14 +450,55 @@ const getSelectPupilNotTakingCheck = async (req, res, next) => {
 
   return res.render('school/select-pupils-not-taking-check', {
     breadcrumbs: req.breadcrumbs(),
+    sortField,
+    sortDirection,
     attendanceCodes,
-    formData,
     pupilsList,
     htmlSortDirection,
     arrowSortDirection
   })
 }
 
+const savePupilNotTakingCheck = async (req, res, next) => {
+  res.locals.pageTitle = 'Save pupils not taking the check'
+  req.breadcrumbs(res.locals.pageTitle)
+
+  if (req.body.attendanceCode === undefined || req.body.pupil === undefined) {
+    return res.redirect('/school/pupils-not-taking-check/select-pupils')
+  }
+
+  const todayDate = moment(moment.now()).format()
+  const pupilsData = await fetchMultiplePupils(Object.values(req.body.pupil))
+  let pupilsToSave = []
+
+  await pupilsData.map(async (pupil) => {
+    if (pupil) {
+      pupil.attendanceCode = {
+        _id: req.body.attendanceCode,
+        dateRecorded: new Date(todayDate),
+        byUser: req.user.UserName
+      }
+      await pupilsToSave.push(pupil)
```

Comment:
I don't think you need to await on an `Array.push()`.  This whole block should be synchronous.